### PR TITLE
SPOI-11327: StramEvents are grouping correction, to remove processed …

### PR DESCRIPTION
**Pull request by DT-Priyanka**
…requests

### CODE REVIEW

**Comment by DT-Priyanka** [comment]
@sanjaypujare  please review and merge.

**Comment by sanjaypujare** [comment]
There is no code comment, JIRA comment or PR comment that describes where exactly the bug is and how the code change fixes the problem

File: engine/src/main/java/org/apache/apex/stram/DeployManager.java [hunk 1]
```diff
@@ -155,18 +157,21 @@ private boolean removeOperatorFromDeployRequest(int operatorId)
   }
 
   /*
-   * Remove deployRequest from StrAM if it has no more pending operators to deploy
+   * Remove deployRequests from StrAM if it has no more pending operators to deploy i.e. requests has been fully processed
    * @param containerId
```
Comment (sanjaypujare) [inline engine/src/main/java/org/apache/apex/stram/DeployManager.java]: A more descriptive comment would be useful e.g. "remove all the fully processed deployRequests the containerId belongs. A deployRequests is fully processed if it has no more pending operators to deploy"

File: engine/src/main/java/org/apache/apex/stram/DeployManager.java [hunk 2]
```diff
@@ -155,18 +157,21 @@ private boolean removeOperatorFromDeployRequest(int operatorId)
...
    * @return isRemoved
    */
-  private boolean removeProcessedDeployRequest(String containerId)
+  private void removeProcessedDeployRequests(String containerId)
```
Comment (sanjaypujare) [inline engine/src/main/java/org/apache/apex/stram/DeployManager.java]: The parameter containerId is not even used but the previous version used it (although it stopped after removing the first deployRequest). This is potentially a serious bug or an inconsistency in the design/implementation of the feature that needs to be explained.

File: engine/src/test/java/org/apach/apex/stram/DeployManagerTest.java [hunk 3]
```diff
@@ -142,12 +142,12 @@ public void testAddNewContainerToDeployRequest()
   @Test
```
Comment (sanjaypujare) [inline engine/src/test/java/org/apach/apex/stram/DeployManagerTest.java]: Can you add test that simulates the bug found and demonstrates how the code change fixes it? e.g. create a launch time deploy list, then simulate a failures and show that there are multiple deployreqeusts with the same containerId

File: engine/src/main/java/org/apache/apex/stram/DeployManager.java (hunk 2 above)
Comment (DT-Priyanka) [inline engine/src/main/java/org/apache/apex/stram/DeployManager.java]: previously I was trying to remove request for containerId now I am trying to remove requests which are completely processed i.e. they don't have any pending deploy. I will remove the parameter.

File: engine/src/test/java/org/apach/apex/stram/DeployManagerTest.java (hunk 3 above)
Comment (DT-Priyanka) [inline engine/src/test/java/org/apach/apex/stram/DeployManagerTest.java]: We already have a test "testRemoveProcessedDeployRequest" it did not fail earlier as I had put the containerId of operator itself. So our deploy request for the given containerId.
Ideally we would like to deploy request irrespective of containerId if it's completely processed. I have changed "failedContainerId" so the test actually fails without code changes. And works with code changes.

**Comment by DT-Priyanka** [comment]
I would close this one and open new PR #5 to use right branch name and commit message. Please refer #5  for further review and merge process.

File: engine/src/main/java/org/apache/apex/stram/DeployManager.java [hunk 4]
```diff
@@ -157,11 +157,10 @@ private boolean removeOperatorFromDeployRequest(int operatorId)
   }
 
   /*
-   * Remove deployRequests from StrAM if it has no more pending operators to deploy i.e. requests has been fully processed
-   * @param containerId
+   * Remove all the fully processed deployRequests. A deployRequests is fully processed if it has no more pending operators to deploy
    * @return isRemoved
    */
-  private void removeProcessedDeployRequests(String containerId)
```
Comment (sanjaypujare) [inline engine/src/main/java/org/apache/apex/stram/DeployManager.java]: As discussed we need to check both lists are empty (deployList and undeployList). So there is still a bug there.

**Pull request closed without merging**